### PR TITLE
Add "raw" `sizemode` to cone trace

### LINKED ISSUE #3613: Actually absolute sizemode for cones
Issue originally posted at https://github.com/plotly/plotly.py/issues/1443

If I understand the documentation correctly there is no way to manually scale the cones in the plot units. Even when using sizemode='absolute' there is some magic rescaling of the length of the cones.
I would expect a way to use the norm of the (u,v,w) vector directly as the length of the cones, but this is not really possible unless you understand how the scaling factor is computed.

### PR DESCRIPTION
**Pull request by archmoj**
Resolves #3613.

@plotly/plotly_js 

[demo](https://codepen.io/MojtabaSamimi/pen/zYXZMPo?editors=0010)

### CODE REVIEW

File: stackgl_modules/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "box-intersect": "plotly/box-intersect#v1.1.0",
     "convex-hull": "^1.0.3",
     "delaunay-triangulate": "^1.1.6",
-    "gl-cone3d": "^1.5.2",
+    "gl-cone3d": "github:gl-vis/gl-cone3d#9cf07db8d3526dbe85e9fa671857416cc3ed63e2",
```
Comment (archmoj) [inline stackgl_modules/package.json:16]: https://github.com/gl-vis/gl-cone3d/pull/20

File: src/traces/cone/attributes.js [hunk 2]
```diff
@@ -96,13 +96,14 @@ var attrs = {
 
     sizemode: {
         valType: 'enumerated',
-        values: ['scaled', 'absolute'],
+        values: ['scaled', 'absolute', 'vector'],
```
Comment (archmoj) [inline src/traces/cone/attributes.js:99]: Another option instead of calling it *vector* is *uvw* which refers to the data arrays the size is coming from.

**Comment by Sharpz7** [comment]
This looks great. I tried to test it locally, but was unable to get the `python setup.py updateplotlyjs` with `"plotly.js": "github:plotly/plotly.js#5808be251ef2c41252973a48431c99a34e1c1d8c"` to run, but I trust that this is implemented correctly.

I think a name like "constant-absolute" makes more sense from a end-user perspective. To be honest, I assumed that "absolute" would give me what I want - ignoring the internal factor and just using the value I give.

A name, or another alias that better gets that across would be great.

**Review by Sharpz7 — APPROVED** [review]

**Comment by alexcjohnson** [comment]
@Sharpz7 good point - TBH I’ve forgotten the rationale behind the original two modes and their names, but if we were designing the API today, the new version would probably have been called `absolute`.  But we have a bit of a conundrum since that’s already taken. 

@archmoj i haven’t played with this much, but if I’m understanding it correctly, `absolute` scales all the sizes up or down so they don’t intersect, leaving the ratios between different sized cones unchanged. Whereas `scaled` makes the largest cone the same size as `absolute` but the smaller ones may be larger than they would have been, so that you can still see their directions even if their real magnitudes are much smaller, is that right?

If I have that right, I might suggest `raw` as the name for the new one. That’s about the only word I can think of that implies “even less processed than absolute”

File: stackgl_modules/package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "box-intersect": "plotly/box-intersect#v1.1.0",
     "convex-hull": "^1.0.3",
     "delaunay-triangulate": "^1.1.6",
-    "gl-cone3d": "^1.5.2",
+    "gl-cone3d": "github:gl-vis/gl-cone3d#846764e2c95157fcfcf1ef264401c3cf665faaaa",
```
Comment (archmoj) [inline stackgl_modules/package.json]: https://github.com/gl-vis/gl-cone3d/pull/20

**Pull request merged**